patientIdentifierType.md

```diff
@@ -0,0 +1,104 @@
+
+* #### List patientIdentifierType
+
+    Fetch all non-retired patientIdentifierTypes resources that match any specified parameters otherwise fetch all non-retired patients. Returns a `200 OK` status with the patientIdentifierType response. If the user is not logged in it returns `401 Unauthorized` status is returned.
+
+    ```console
+    GET /patientidentifiertype?
+        &q=**searchqQuery**
+    ```
+
+* #### List patientIdentifierType by UUID.
+
+    Retrieve a patientIdentifierType by its UUID. Returns a `404 Not Found` status if patientIdentifierType does not exist in the system. If the user is not logged in to perform this action, a `401 Unauthorized` status is returned.
+
+    ```console
+    GET /patientidentifiertype/:target_patientIdentifierType_uuid
+    ```
+### Create a patientIdentifierType
+
+* To create a patientIdentifierType you need to specify the below properties in the request. If you are not logged in to perform this action, a `401 Unauthorized` status is returned.
+
+    #### Properties
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *name* | string | label for the identifier
+    *description* | string | a small description about the patientIdentifier
+    *format* | string | a regular expression defining what the identifier text should contain
+    *formatDescription* | string | an optional description of the regular expression
+    *required* | boolean | a true/false whether every patient MUST have this type
+    *checkDigit* | boolean | a true/false whether this identifier has a checkdigit at the end
+    *validator* | string | `org.openmrs.patient.IdentifierValidator`
+    *locationBehavior* | "REQUIRED" or "NOT USED" | behavior of the location with respect to the identifier 
+    *uniquenessBehavior* | string | specify the uniqueness of the behaviour, it can be either Unique, Non Unique or Location.
+
+    ```console
+    POST /patientidentifiertype
+    {
+        "name": "Wilson Hospital Medical Record Number",
+        "description": "Wilson Hospital Medical Record Number is used as a patientIdentifier type.",
+        "format": "234-7",
+        "formatDescription": "Must follow the pattern NNN-N",
+        "required": true,
+        "checkDigit": true,
+        "locationBehavior": "REQUIRED",
+        "uniquenessBehavior": "Unique"
+    }
+    ```
+### Update a patientIdentifierType
+
+* Update a target patientIdentifierType with given UUID, this method only modifies properties in the request. 
+Returns a `404 Not Found` status if patientIdentifierType not exists. If user not logged in to perform this action, a `401 Unauthorized status returned`.
+
+    #### Properties
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *Object* | `PatientIdentifierType` | Patient resource with updated properties (same body as while creating the resource body)
+
+    ```console
+    POST /patientidentifertype/:target_patientidentifiertype_uuid
+    -d modified_patientidentifiertype_object
```

Comment:
I'm not sure what this means. I assume the `-d` is a reference to a [cURL](https://en.wikipedia.org/wiki/CURL) command, suggesting the updates would be in JSON format in a file named "modified_patientidentifiertype_object". Please omit the `-d` option and include an example JSON object after the POST command. See [encounter.md](https://github.com/openmrs/openmrs-contrib-rest-api-docs/blob/be77aa2b681039a9affadb291552a68533ab003e/Encounter/encounter.md) for an example.

Reply:
@bmamlin for this particular object I'll use this one  : 
` {
        "name": "Wilson Hosp MRN",

        "description": "Wilson Hospital Medical Record Number",

        "format": "\d{1,10}-\d",

        "formatDescription": "Up to ten digts followed by a hyphen and another digit",

        "required": true,

        "validator": "org.openmrs.patient.impl.LuhnIdentifierValidator",

        "checkDigit": true,

        "locationBehavior": "REQUIRED",

        "uniquenessBehavior": "Unique"

    }`